base/rts/rts.c

```diff
@@ -2792,10 +2792,37 @@ int main(int argc, char **argv) {
 #ifdef ACTON_THREADS
     cpu_set_t cpu_set;
 
+    size_t primary_thread_stack_size;
+    size_t target_thread_stack_size;
+#if defined(IS_MACOS)
+    primary_thread_stack_size = pthread_get_stacksize_np(pthread_self());
+#else
+    pthread_attr_t attr;
+    pthread_getattr_np(pthread_self(), &attr);
+    pthread_attr_getstacksize(&attr, &primary_thread_stack_size);
+    pthread_attr_destroy(&attr);
+#endif
+    target_thread_stack_size = REQUIRED_STACK_SIZE > primary_thread_stack_size ? (size_t)REQUIRED_STACK_SIZE : primary_thread_stack_size;
+
+    if (primary_thread_stack_size < target_thread_stack_size)
+        log_warn("Current primary thread stack size: %u, required thread stack size: %u", primary_thread_stack_size, target_thread_stack_size);
+
+    pthread_attr_t ss_attr;
+    size_t secondary_thread_stack_size = 0;
+    pthread_attr_init(&ss_attr);
+    pthread_attr_getstacksize(&ss_attr, &secondary_thread_stack_size);
+    if (secondary_thread_stack_size < target_thread_stack_size)
+    {
+        log_debug("Secondary thread stack size: %d, required thread stack size: %d", secondary_thread_stack_size, target_thread_stack_size);
+        int err = pthread_attr_setstacksize(&ss_attr, target_thread_stack_size);
+        if (err)
+            log_error("pthread_attr_setstacksize failed: %s", strerror(err));
+    }
+
     // RTS Monitor Log
     pthread_t mon_log_thread;
     if (mon_log_path) {
-        pthread_create(&mon_log_thread, NULL, $mon_log_loop, (void *)(intptr_t)mon_log_period);
+        pthread_create(&mon_log_thread, &ss_attr, $mon_log_loop, (void *)(intptr_t)mon_log_period);
```

Comment:
Just FYI, the RTS monitor thread do not run Acton continuations, only some functions written in C, so they really don't require a large stack space. Though it certainly doesn't hurt either.

I'd actually like to remove the RTS mon thread overall...